### PR TITLE
Add release notes for SimPEG v0.21

### PR DESCRIPTION
**Pull request by santisoler**
#### Summary

Add release notes for SimPEG v0.21.0 to the docs.

#### PR Checklist
* [ ] If this is a work in progress PR, set as a Draft PR
* [ ] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/getting_started/contributing/code-style.html).
* [ ] Added [tests](https://docs.simpeg.xyz/content/getting_started/practices.html#testing) to verify changes to the code.
* [ ] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/getting_started/practices.html#documentation).
* [ ] Marked as ready for review (if this is was a draft PR), and converted 
      to a Pull Request
* [ ] Tagged ``@simpeg/simpeg-developers`` when ready for review.

### CODE REVIEW

**Comment by santisoler** [comment]
@jcapriot @lheagy would you like to take a look at these release notes?

File: docs/content/release/0.21.0-notes.rst [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+New features
+------------
+
+Gravity simulation using Choclo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now we can use a faster and more memory efficient implementation of the gravity
+simulation ``SimPEG.potential_fields.gravity.Simulation3DIntegral``, making use
+of Choclo and Numba.
```
Comment (jcapriot) [inline docs/content/release/0.21.0-notes.rst:20]:
```suggestion
of Choclo and Numba.  To make use of this functionality you will need to install `choclo` in addition to `simpeg`.
```

File: docs/content/release/0.21.0-notes.rst [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+use Dask with ``SimPEG.meta.MetaSimulations``.
+
+See https://github.com/simpeg/simpeg/pull/1199.
+
+Rotated Gradients
+~~~~~~~~~~~~~~~~~
+
+Added a new ``SimPEG.regularization.SmoothnessFullGradient`` regularization
+class that allows to regularize first order smoothness along any arbitrary
+direction, enabling anisotropic weighting.
```
Comment (jcapriot) [inline docs/content/release/0.21.0-notes.rst:37]:
```suggestion
direction, enabling anisotropic weighting. This regularization also works for a `SimplexMesh`.
```

File: docs/content/release/0.21.0-notes.rst [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+See https://github.com/simpeg/simpeg/pull/1167.
+
+Logistic Sigmoid Map
+~~~~~~~~~~~~~~~~~~~~
+
+New ``SimPEG.map.LogisticSigmoidMap`` mapping class that computes the logistic
+sigmoid of the model parameters.
```
Comment (jcapriot) [inline docs/content/release/0.21.0-notes.rst:45]:
```suggestion
sigmoid of the model parameters. This is an alternative method to incorporate upper and lower bounds on model parameters.
```

File: docs/content/release/0.21.0-notes.rst [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+See https://github.com/simpeg/simpeg/pull/1352.
+
+Store sensitivity matrix in NSEM and FDEM simulations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Allow to store the sensitivity matrix in natural source and frequency domain
+EM simulations through the ``getJ`` method.
```
Comment (jcapriot) [inline docs/content/release/0.21.0-notes.rst:53]:
```suggestion
Create Jacobian matrix in NSEM and FDEM simulations
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The frequency domain electromagnetic simulations (including natural source) now support creating and storing the Jacobian matrix. You can access it by using the ``getJ`` method.
```

File: docs/content/release/0.21.0-notes.rst [hunk 5]
```diff
@@ -0,0 +1,266 @@
+.. _0.21.0_notes:
+
+===========================
+SimPEG 0.21.0 Release Notes
+===========================
+
+April 8th, 2024
+
```
Comment (jcapriot) [inline docs/content/release/0.21.0-notes.rst:8]: Consider adding a TOC at the top,

```
.. contents:: Highlights
    :depth: 3
```
or whichever depth makes most sense.

**Review by jcapriot — CHANGES_REQUESTED** [review]

File: docs/content/release/0.21.0-notes.rst (hunk 1 above)
Comment (santisoler) [reply docs/content/release/0.21.0-notes.rst:20]: Totally! Nice catch.

File: docs/content/release/0.21.0-notes.rst [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+------------
+
+Gravity simulation using Choclo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now we can use a faster and more memory efficient implementation of the gravity
+simulation ``SimPEG.potential_fields.gravity.Simulation3DIntegral``, making use
+of Choclo and Numba.  To make use of this functionality you will need to
+`install ``choclo`` <https://www.fatiando.org/choclo/latest/install.html>`__ in
+addition to `SimPEG`.
```
Comment (santisoler) [inline docs/content/release/0.21.0-notes.rst:25]:
```suggestion
addition to ``SimPEG``.
```

File: docs/content/release/0.21.0-notes.rst [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+===========================
+SimPEG 0.21.0 Release Notes
+===========================
+
+April 8th, 2024
+
+.. contents:: Highlights
+    :depth: 3
+
+Updates
```
Comment (lheagy) [inline docs/content/release/0.21.0-notes.rst:12]: I believe that the SP simulation is also new in this release, is that right?

Comment (lheagy) [reply docs/content/release/0.21.0-notes.rst:12]: Ah, nevermind, it looks like this was in 0.20
https://github.com/simpeg/simpeg/pull/1211

**Review by lheagy — APPROVED** [review]
This looks good to me, thanks for putting it together @santisoler, and for the review @jcapriot! It will be great to get this out into the world 🚀

**Pull request merged**